configure.ac

```diff
@@ -3070,7 +3070,7 @@ if test x$host_win32 = xno; then
 	])
 
 	AC_MSG_CHECKING(for sendfile with 6 arguments)
-	AC_TRY_COMPILE([
+	AC_TRY_LINK([
```

Comment:
How can you tell 4 parameter vs. 6 parameter? They will each compile and each link. You should compile the test program as C++ to decide. See what I did with size_t.

Reply:
The four-argument test will fail on not finding the sendfile header. The six-argument test will *compile* due to implied declaration, (it uses headers found on mostly any system) but linking will make it fail, which is what we'd want.

Reply:
Yeah I see that now..on Mac..	#include <sys/sendfile.h>
 not found. Wierd.

But Linux appears to have sys/sendfile and 4 parameters.

But then, the thing being compiled..the number of parameters is irrelevant. You can just take the address of sendfile in a global.

It is really testing if sys/sendfile exists, and if so, assume it has 4 parameters (if it doesn't, it'll just warn and succeed).
Or else, does sendfile link, in which case, assume 6 parameters.

There isn't really a check for number of parameters. It is just that sys/sendfile.h implies 4 parameters. 

It isn't really doing  what it looks like.

Reply:
Suggest at least comments. That the second check will always compile, and if it links, the signature is merely assumed. There is no actual checking of the signature.

And that the first test will compile if sys/sendfile.h exists, no matter what is the signature or existance of sendfile.

Reply:
FWIW, if sendfile-with-4 is caught, the PAL defines IIRC wont fall back to sendfile-with-6; either way, this patch wouldn't change *that* behaviour.

Reply:
Indeed, it was already wonky. You always get one of the symbols, sometimes both, must check them in order, this doesn't change it. It is a minor bug that it defines both, that we must workaround.